### PR TITLE
Update fiber photometry pipeline

### LINKED ISSUE #25: update or pin `ndx-photometry`
After https://github.com/catalystneuro/ndx-photometry/pull/7 is merged I'll either have to update this one line in the fiber photometry folder so the conversion won't break.

```
fibers_table.add_fiber(
        excitation_sources=[0],
        ...
)
```

The alternative is to pin `ndx-photometry==0.2.0`

### LINKED ISSUE #37: TypeError: Can't instantiate abstract class FiberPhotometryInterface with abstract method add_to_nwbfile
I am testing out the fiber_photometry module for the first time.  I have built my conda environment "tye_lab_to_nwb" with the following commands:
```
conda activate tye_lab_to_nwb
pip install -e .
pip install -r src/tye_lab_to_nwb/fiber_photometry/requirements.txt
```
I update the convert_session.py to point to the correct photometry_file_path and output_dir_path locations.  Then I run `python src/tye_lab_to_nwb/fiber_photometry/convert_session.py` and I get the following error:
```
(tye_lab_to_nwb_env) lkeyes@node32:~/Projects/GIT/tye-lab-to-nwb$ python src/tye_lab_to_nwb/fiber_photometry/convert_session.py
Traceback (most recent call last):
  File "/nadata/snlkt/home/lkeyes/Projects/GIT/tye-lab-to-nwb/src/tye_lab_to_nwb/fiber_photometry/convert_session.py", line 50, in <module>
    session_to_nwb(
  File "/nadata/snlkt/home/lkeyes/Projects/GIT/tye-lab-to-nwb/src/tye_lab_to_nwb/fiber_photometry/convert_session.py", line 16, in session_to_nwb
    interface = FiberPhotometryInterface(file_path=str(file_path))
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: Can't instantiate abstract class FiberPhotometryInterface with abstract method add_to_nwbfile

```

### PR DESCRIPTION
**Pull request by weiglszonja**
Fix #37 to work in updated neuroconv environment
resolve #25 by pinning `ndx-photometry==0.2.0`
- pins hdmf==3.6.1 until https://github.com/catalystneuro/ndx-photometry/issues/8 is resolved
- add [convert_all_sessions.py ](https://github.com/catalystneuro/tye-lab-to-nwb/blob/cf023a5815f79077e3a101715f16142ecf972e69/src/tye_lab_to_nwb/fiber_photometry/convert_all_sessions.py)
- add [fiber_photometry_notes.md](https://github.com/catalystneuro/tye-lab-to-nwb/blob/cf023a5815f79077e3a101715f16142ecf972e69/src/tye_lab_to_nwb/fiber_photometry/fiber_photometry_notes.md)

### CODE REVIEW

**Comment by weiglszonja** [comment]
Usage example
```
conda activate tye_lab_to_nwb
pip install -r src/tye_lab_to_nwb/fiber_photometry/requirements.txt
python src/tye_lab_to_nwb/fiber_photometry/convert_session.py
```
or parallel convert with a session_config.xlsx file as described in [fiber_photometry_notes.md](https://github.com/catalystneuro/tye-lab-to-nwb/blob/cf023a5815f79077e3a101715f16142ecf972e69/src/tye_lab_to_nwb/fiber_photometry/fiber_photometry_notes.md)
```
conda activate tye_lab_to_nwb
pip install -r src/tye_lab_to_nwb/fiber_photometry/requirements.txt
python src/tye_lab_to_nwb/fiber_photometry/convert_all_sessions.py
```

**Comment by CodyCBakerPhD** [comment]
As mentioned, will need to add a precise commit pin within the neuroconv branch https://github.com/catalystneuro/neuroconv/pull/579 to avoid issues with our new minimal requirement of `hdmf>=3.10.0`: I recommend `36fe8ba02c036e715a915ee66079391a50536fb2`

File: src/tye_lab_to_nwb/neurotensin_valence/neurotensin_valence_requirements.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
--e git+https://github.com/catalystneuro/neuroconv@add_ignore_timestamps_errors_to_interface#egg=neuroconv
+git+https://github.com/catalystneuro/neuroconv.git@36fe8ba02c036e715a915ee66079391a50536fb2#egg=neuroconv
```
Comment (weiglszonja) [inline src/tye_lab_to_nwb/neurotensin_valence/neurotensin_valence_requirements.txt:1]: @CodyCBakerPhD I added here, should I also add to `fiber_photometry` requirements as well?

Comment (CodyCBakerPhD) [reply src/tye_lab_to_nwb/neurotensin_valence/neurotensin_valence_requirements.txt:1]: @weiglszonja Probably; in any situation where there is a combination of `main` of `neuroconv` and `ndx-fiber-photometry`

Comment (weiglszonja) [reply src/tye_lab_to_nwb/neurotensin_valence/neurotensin_valence_requirements.txt:1]: Added there too until https://github.com/catalystneuro/ndx-photometry/issues/8 is resolved (Ryan is on it!)

Comment (weiglszonja) [reply src/tye_lab_to_nwb/neurotensin_valence/neurotensin_valence_requirements.txt:1]: good to go @CodyCBakerPhD

**Review by CodyCBakerPhD — APPROVED** [review]

**Pull request merged**